tests/php/test-tag-and-attribute-sanitizer.php

```diff
@@ -2715,6 +2745,30 @@ class="slide"
 				null,
 				[ 'amp-redbull-player' ],
 			],
+
+			'mustache_templates_with_variable_attrs'       => [
+				'
+				<template type="amp-mustache">
+					<!--🚫--><amp-img src="/img1.png" layout="bad" width="bad" height="bad"></amp-img>
+					<!--✅--><amp-img src="/img2.png" layout="{{layout}}" width="bad" height="bad"></amp-img>
+					<!--✅--><amp-img src="/img3.png" layout="bad" width="{{width}}" height="bad"></amp-img>
+					<!--✅--><amp-img src="/img4.png" layout="bad" width="bad" height="{{height}}"></amp-img>
+					<!--✅--><amp-img src="/img5.png" layout="bad" width="bad" height="bad" heights="{{height}}"></amp-img>
+					<!--✅--><amp-img src="/img6.png" layout="bad" width="bad" height="bad" heights="bad" sizes="{{sizes}}"></amp-img>
```

Comment:
These are all valid in the AMP validator. But the the sanitizer is removing _some_ of the attributes here.

See comment: https://github.com/ampproject/amp-wp/pull/4583#discussion_r408507385

> I realize that the above sample should probably rather (or in addition) be added to the tag-and-attribute tests because this is where the mustache template vars are being checked, right? When doing so, it shows a difference of behavior from the AMP Validator. See 891c108651faf70d49c216e477de3ae7734fe395.
>
> Now, it could be argued that the AMP plugin is actually doing something better because it is removing the attributes that could never be valid even when Mustache template handling is being performed. If that is the case, we should codify the behavior in this `mustache_templates_with_variable_attrs` test case to show how it behaves.

Reply:
It's currently getting sanitized as:

```html
<!--✅--><amp-img src="/img2.png" layout="{{layout}}" width="bad" height="bad"></amp-img>
<!--✅--><amp-img src="/img3.png" width="{{width}}" height="bad"></amp-img>
<!--✅--><amp-img src="/img4.png" width="bad" height="{{height}}"></amp-img>
<!--✅--><amp-img src="/img5.png" width="bad" height="bad" heights="{{height}}"></amp-img>
<!--✅--><amp-img src="/img6.png" width="bad" height="bad" heights="bad" sizes="{{sizes}}"></amp-img>
```